### PR TITLE
chore: add dependabot config

### PR DESCRIPTION
**Pull request by armano2**
currently we have bunch of out of date dependencies, this configuration will exclude unwanted eslint and typescript updates as changes require manual work

to my knowledge dependabot reports now only updates that have security updates in them.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Please see the documentation for all configuration options:
+# https://help.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+  - package-ecosystem: "npm"
+    directory: "/"
+    schedule:
+      interval: "weekly"
+      day: "monday"
```
Comment (armano2) [inline .github/dependabot.yml:12]: not sure how often we want it to be trigger

Comment (bradzacher) [reply .github/dependabot.yml:12]: Lets start with weekly. If it ends up creating too much noise each week, we can always bump it to monthly.

**Review by bradzacher — APPROVED** [review]

**Comment by armano2** [comment]
@bradzacher small note, rn, because we have alot of out of date dependencies, we can get a little spammed

https://github.com/typescript-eslint/typescript-eslint/network/updates/88363309

**Comment by bradzacher** [comment]
yeah I saw that! I also saw the limit was 5 (I was worried it might spam us!)

I figured we should probably just do a one-time "bump everything" manual commit and then let dependabot start doing its incremental thing next week.

can take a look on the weekend.

**Comment by armano2** [comment]
limit of 5 is a default one, it can be adapted/changed

**Pull request merged**